src/java.base/share/classes/java/text/DecimalFormat.java

```diff
@@ -2575,7 +2648,24 @@ int subparseNumber(String text, int position,
             }
         }
         return position;
+    }
+
+    // Checks to make sure grouping size is not violated. Used when strict.
```

Comment:
If it is only supposed to be used in `strict`, might be helpful to put an assertion here.